cloudsdk/src/test/java/io/particle/android/sdk/cloud/TestData.kt

```diff
@@ -0,0 +1,107 @@
+package io.particle.android.sdk.cloud
+
+import io.particle.android.sdk.cloud.ParticleDevice.ParticleDeviceType
+
+const val DEVICE_ID_0 = "d34db33f52ca40bd34db33f0"
+const val DEVICE_ID_1 = "d34db33f52ca40bd34db33f1"
+
+internal val DEVICE_STATE_0 = DeviceState(
```

Comment:
This is a step in a right direction, but I would also look into extracting that into stand alone json files. If we ever come up with a nice way to record these from API automatically, test cases would be ready for that change :). Not sure if worth changing now (not sure if I'm doing that for iOS now), but we might think about it until we have too much stuff hardcoded.

Reply:
I don't think I want to move to pure JSON files, but I just found [a good VCR equivalent for Java](https://github.com/airbnb/okreplay), which I'd be up for moving to in Q1.

Moving your tests to [DVR](https://github.com/venmo/DVR) requires that the SDK be rewritten in Swift, correct?

Reply:
Unfortunately, yes :( We rely on the lib in obj-c and it's not compatible with DVR (or any of the similar libs). I've seen that we could use same lib in swift (same creator, same concept, different lib) and most popular DVR implementation supports that lib.